src/main/java/org/tools4j/hoverraft/state/TrackedFollowerState.java

```diff
@@ -45,4 +45,21 @@ public long nextIndex() {
     public long matchIndex() {
         return matchIndex;
     }
+
+    void nextIndex(final long index) {
+        this.nextIndex = index;
+    }
+
+    void decrementNextIndex() {
+        this.nextIndex--;
+    }
+
+    void resetMatchIndex() {
+        this.matchIndex = 0;
```

Comment:
the paper says matchIndex should be reset to 0 on new leadership. That could mean -1 if indexing starts with 0.